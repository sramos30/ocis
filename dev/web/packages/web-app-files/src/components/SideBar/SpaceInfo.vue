<template>
  <div class="space_info">
    <div class="space_info__body oc-text-overflow oc-flex oc-flex-middle">
      <div class="oc-mr-s">
        <oc-icon
          name="layout-grid"
          :size="space.description ? 'large' : 'medium'"
          class="oc-display-block"
        />
      </div>
      <div>
        <h3 data-testid="space-info-name" v-text="space.name" />
        <span data-testid="space-info-subtitle" v-text="space.description" />
      </div>
    </div>
  </div>
</template>

<script>
import Mixins from '../../mixins'

export default {
  name: 'SpaceInfo',
  mixins: [Mixins],
  inject: ['displayedItem'],
  computed: {
    space() {
      return this.displayedItem.value
    }
  }
}
</script>

<style lang="scss">
.space_info {
  display: grid;
  grid-template-columns: auto 1fr auto;
  align-items: center;
  grid-gap: 5px;

  &__body {
    text-align: left;
    font-size: 0.75rem;

    h3 {
      font-size: 0.9rem;
      font-weight: 600;
      margin: 0;
    }
  }
}
</style>
