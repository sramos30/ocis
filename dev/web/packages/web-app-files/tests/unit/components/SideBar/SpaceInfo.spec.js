import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import GetTextPlugin from 'vue-gettext'
import AsyncComputed from 'vue-async-computed'

import stubs from '@/tests/unit/stubs'

import SpaceInfo from '@files/src/components/SideBar/SpaceInfo.vue'

const spaceMock = {
  type: 'space',
  name: ' space',
  id: '1',
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  spaceQuota: {
    used: 100
  }
}

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(AsyncComputed)
localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

const selectors = {
  name: '[data-testid="space-info-name"]',
  subtitle: '[data-testid="space-info-subtitle"]'
}

const formDateFromRFC = jest.fn()
const formRelativeDateFromRFC = jest.fn()
const resetDateMocks = () => {
  formDateFromRFC.mockReset()
  formRelativeDateFromRFC.mockReset()
  formDateFromRFC.mockImplementation(() => 'ABSOLUTE_TIME')
  formRelativeDateFromRFC.mockImplementation(() => 'RELATIVE_TIME')
}

describe('SpaceInfo', () => {
  it('shows space info', () => {
    resetDateMocks()

    const wrapper = createWrapper(spaceMock)
    expect(wrapper.find(selectors.name).exists()).toBeTruthy()
    expect(wrapper.find(selectors.subtitle).exists()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
})

function createWrapper(spaceResource) {
  return shallowMount(SpaceInfo, {
    store: new Vuex.Store({
      getters: {
        user: function () {
          return { id: 'marie' }
        },
        capabilities: jest.fn(() => ({}))
      },
      modules: {
        Files: {
          namespaced: true,
          getters: {
            highlightedFile: function () {
              return spaceResource
            }
          }
        }
      }
    }),
    localVue,
    stubs: {
      ...stubs,
      'oc-resource-icon': true,
      'oc-resource-name': true
    },
    directives: {
      OcTooltip: null
    },
    mixins: [
      {
        methods: {
          formDateFromRFC,
          formRelativeDateFromRFC
        }
      }
    ],
    mocks: {
      $router: {
        currentRoute: {
          name: 'some-route',
          query: { page: 1 }
        },
        resolve: (r) => ({ href: r.name })
      },
      publicPage: () => false
    },
    provide: {
      displayedItem: {
        value: spaceResource
      }
    }
  })
}
