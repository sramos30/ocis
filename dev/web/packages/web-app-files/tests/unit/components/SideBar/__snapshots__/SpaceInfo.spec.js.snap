// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceInfo shows space info 1`] = `
<div class="space_info">
  <div class="space_info__body oc-text-overflow oc-flex oc-flex-middle">
    <div class="oc-mr-s">
      <oc-icon-stub name="layout-grid" size="medium" class="oc-display-block"></oc-icon-stub>
    </div>
    <div>
      <h3 data-testid="space-info-name"> space</h3> <span data-testid="space-info-subtitle"></span>
    </div>
  </div>
</div>
`;
